src/main/java/com/iopipe/plugin/eventinfo/EventInfoPlugin.java

```diff
@@ -0,0 +1,120 @@
+package com.iopipe.plugin.eventinfo;
+
+import com.iopipe.IOpipeConstants;
+import com.iopipe.IOpipeExecution;
+import com.iopipe.plugin.IOpipePlugin;
+import com.iopipe.plugin.IOpipePluginExecution;
+import com.iopipe.plugin.IOpipePluginPostExecutable;
+import com.iopipe.plugin.IOpipePluginPreExecutable;
+
+/**
+ * This is the trace plugin which is used to track specific marks and measure
+ * performance.
+ *
+ * @since 2018/04/22
+ */
+public class EventInfoPlugin
+	implements IOpipePlugin, IOpipePluginPostExecutable,
+		IOpipePluginPreExecutable
+{
+	/** Default set of decoders, lazily initialized. */
+	private volatile EventInfoDecoders _decoders;
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public boolean enabledByDefault()
+	{
+		return true;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public IOpipePluginExecution execute(IOpipeExecution __e)
+	{
+		if (__e == null)
+			throw new NullPointerException();
+		
+		// Lazily initialize decoders since the plugin might not actually
+		// be enabled
+		EventInfoDecoders decoders = this._decoders;
+		if (decoders == null)
+			this._decoders = (decoders = new EventInfoDecoders());
+		
+		return new EventInfoExecution(__e, decoders);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public Class<? extends IOpipePluginExecution> executionClass()
+	{
+		return EventInfoExecution.class;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public String homepage()
+	{
+		return "https://github.com/iopipe/iopipe-java";
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public String name()
+	{
+		return "event-info";
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/23
+	 */
+	@Override
+	public void postExecute(IOpipePluginExecution __e)
+		throws NullPointerException
+	{
+		if (__e == null)
+			throw new NullPointerException();
+		
+		((EventInfoExecution)__e).__post();
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/23
+	 */
+	@Override
+	public void preExecute(IOpipePluginExecution __e)
+		throws NullPointerException
+	{
+		if (__e == null)
+			throw new NullPointerException();
+		
+		((EventInfoExecution)__e).__pre();
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/22
+	 */
+	@Override
+	public String version()
+	{
+		return IOpipeConstants.AGENT_VERSION;
```

Comment:
Shouldn't this be its own version? (Update when the plugin updates, not the core agent?) 1.0?

Reply:
Plugin versions and agent versions are independent of each other in other agents. Since an agent release could occur without an update to a plugin. This is helpful in the event there's an issue with a specific plugin/agent version combo.

Reply:
The plugins can be fixed to a specific version, they would all have to be fixed to `1.2.0` so as to not go backwards in the versioning scheme since all the built-in plugin versions match the agent version. All of the core plugins are unit tested so they hopefully should not break at all.

Reply:
Yes, for this PR we could start in the EventInfo plugin, and other plugins could update in a different PR

Reply:
@XerTheSquirrel could you update this to `1.2.0`?